src/Arcus.WebApi.Tests.Runtimes.AzureFunction/HttpTriggerFunction.cs

```diff
@@ -0,0 +1,47 @@
+using System.Threading.Tasks;
+using Arcus.Observability.Correlation;
+using Arcus.WebApi.Logging.Correlation;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Arcus.WebApi.Tests.Runtimes.AzureFunction
+{
+    public class HttpTriggerFunction
+    {
+        private readonly HttpCorrelation _correlationService;
+        private readonly ILogger<HttpTriggerFunction> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpTriggerFunction"/> class.
+        /// </summary>
+        public HttpTriggerFunction(HttpCorrelation correlationService, ILogger<HttpTriggerFunction> logger)
+        {
+            _correlationService = correlationService;
+            _logger = logger;
+        }
+        
+        [FunctionName("HttpTriggerFunction")]
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            if (_correlationService.TryHttpCorrelate(out string errorMessage))
+            {
+                CorrelationInfo correlationInfo = _correlationService.CorrelationInfoAccessor.GetCorrelationInfo();
```

Comment:
Ok now I get your comment, here I'd expect this to be possible:
```suggestion
                CorrelationInfo correlationInfo = _correlationService.GetCorrelationInfo();
```

Which just calls `CorrelationInfoAccessor` under the hood for me.

Reply:
Hmm, no that was actually my recent consideration, that moving the property would maybe make it more 'weird' as we're mixing concepts. I think here with the property makes the 'accessing' and 'determination' of correlation separate.

Reply:
I see it more as an extension and it uses the accessor but it provides more features; but the consumer shouldn't have to care/know that it's using the accessor under the hood imo

Reply:
OK, moved.